src/main/java/eu/enmeshed/EnmeshedOnboardingService.java

```diff
@@ -216,56 +201,37 @@ public RegistrationResult checkRegistrationState(
     }
 
     Relationship relationship = relationships.get(0);
-    RelationshipChange relationshipChange = relationship.getChanges().get(0);
-
-    RelationshipChangeRequestContent relationshipChangeRequestContent =
-        relationshipChange.getRequest().getContent();
-
-    RelationshipCreationChangeRequestContent relationshipCreationChangeRequestContent =
-        (RelationshipCreationChangeRequestContent) relationshipChangeRequestContent;
-
+    RelationshipCreationContent creationContent = relationship.getCreationContent();
     Map<Class<? extends AttributeValue>, AttributeValue> attributes =
-        getSharedSimpleAttributesFromResponseItems(
-            relationshipCreationChangeRequestContent.getResponse().getItems());
+        getSharedSimpleAttributesFromResponseItems(creationContent.getResponse().getItems());
 
-    if (relationshipChange.getStatus() == RelationshipChange.Status.PENDING) {
+    if (relationship.getStatus() == RelationshipStatus.PENDING) {
       boolean decision = acceptanceDecider.test(attributes);
 
       if (decision) {
-        enmeshedClient.acceptRelationshipChange(
-            relationship.getId(),
-            relationshipChange.getId(),
-            ContentWrapper.containing(Collections.emptyMap()));
+        enmeshedClient.acceptRelationship(relationship.getId());
       } else {
-        enmeshedClient.rejectRelationshipChange(
-            relationship.getId(),
-            relationshipChange.getId(),
-            ContentWrapper.containing(Collections.emptyMap()));
+        enmeshedClient.rejectRelationship(relationship.getId());
       }
-
       return checkRegistrationState(relationshipTemplateId, registrationResult -> decision);
-    } else if (relationshipChange.getStatus() == RelationshipChange.Status.ACCEPTED) {
-
+    } else if (creationContent.getResponse().getResult()
```

Comment:
why don't you check this at the relationship and instead at the response inside the creationContent? IMO you will NEVER receive a creationContent where the request was rejected..

Reply:
I changed it to check RelationshipStatus.REJECTED and RelationshipStatus.PENDING status and leave it for eu.enmeshed.model.Response.Result.ACCEPTED as is, because we don't have such a status in  RelationshipStatus. I'm sorry, I adapted to v5 to make our mocks work as soon as possible.

Reply:
I think the status you want to check for there is active.

Reply:
Changed